File: libs/dotnet-sdk/State/FeatureBoardStateSnapshot.cs
using FeatureBoard.DotnetSdk.Models;

namespace FeatureBoard.DotnetSdk.State;

public class FeatureBoardStateSnapshot
{
  private readonly IReadOnlyDictionary<string, FeatureConfiguration> _snapshot;

  public FeatureBoardStateSnapshot(IReadOnlyDictionary<string, FeatureConfiguration> state)
  {
    _snapshot = state;
  }

  public FeatureConfiguration? Get(string featureKey)
  {
    return _snapshot.TryGetValue(featureKey, out var feature)
      ? feature
      : null;
  }
}

File: libs/java-sdk/src/main/java/featureboard/java/sdk/state/FeatureBoardStateSnapshot.java
package featureboard.java.sdk.state;

import featureboard.java.sdk.models.FeatureValue;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class FeatureBoardStateSnapshot {
  private final Map<String, FeatureValue> snapshot;

  public FeatureBoardStateSnapshot(Map<String, FeatureValue> state) {
    snapshot = state;
  }

  public FeatureValue Get(String featureKey) {
    return snapshot.get(featureKey);
  }
}
